webapp/src/main.rs

```diff
@@ -2,84 +2,116 @@
-            }
+    // Backend config loading
+    let mut backend_config: BackendConfig = Figment::new()
+        .merge(Yaml::file("BackendConfig.yaml"))
+        .extract().unwrap();
+    // TODO: cover cases when the path is already full
+    backend_config.sessions_data_dir = current_dir().unwrap()
+        .join(&backend_config.sessions_data_dir);
+    backend_config.sessions_data_file_dir = current_dir().unwrap()
+        .join(&backend_config.sessions_data_file_dir);
 
-        },
-        None => {}
-    }
+    // Logger
+    let decorator = slog_term::TermDecorator::new().build();
+    let drain = slog_term::FullFormat::new(decorator).build().fuse();
+    let drain = slog_async::Async::new(drain).build().fuse();
+    let log = slog::Logger::root(drain, o!());
 
-    match var("COMPILATION_TEMP_DIR") 
+    // Sessions tracker, wrapped in a mutex because it has to be mutable across threads
+    let sessions_tracker;
+    match SessionsTracker::from_file( &backend_config.sessions_data_file_dir)
     {
-        Ok(temp_dir) => 
+        Some(tr) => 
         {
-            println!("COMPILATION_TEMP_DIR already exists.");
-            // Check if the value is a dir path
-            if std::path::Path::new(&temp_dir).is_dir()
-            {
-                return;
-            }
-            else
-            {
-                println!("\"{}\" is not a valid directory path.", temp_dir);
-            }
-        },
-        Err(_) => 
+            info!(log, "Read sessions data from: {:?}", 
+                backend_config.sessions_data_file_dir);
+            sessions_tracker = Arc::new(Mutex::new(tr
+                .life_duration(&Duration::from_millis(backend_config.session_life_duration))));
+        }
+        None => 
         {
-            println!("Can't read COMPILATION_TEMP_DIR.");
+            info!(log, "Couldn't read sessions data from: {:?}", 
+                backend_config.sessions_data_file_dir);
+            sessions_tracker = Arc::new(Mutex::new(SessionsTracker::new()
+                .life_duration(&Duration::from_millis(backend_config.session_life_duration))))
         }
     }
-    // Set COMPILATION_TEMP_DIR with a default value
-    let mut temp_dir = current_dir().unwrap().to_str().unwrap().to_owned();
-    temp_dir.push_str("/tempdata");
 
-    println!("Using default path \"{}\".", temp_dir);
-    set_var("COMPILATION_TEMP_DIR", temp_dir);
+    info!(log, "Backend config:\n {:?}", backend_config);
+
+    rocket::build()
+        .mount("/", FileServer::from("static/"))
+        .mount("/", routes![submit::post_submit])
+        .manage(backend_config)
+        .manage(sessions_tracker)
+        .manage(log)
+        .attach(AdHoc::on_liftoff("Sessions cleaner", |rocket| Box::pin(async move 
+            {
+                let tracker = rocket.state::<Arc<Mutex<SessionsTracker>>>()
+                    .unwrap().to_owned();
+                // TODO: cloning logger here is probably not right, look into it more
```

Comment:
is logger adopted to run in multiple threads?
I suppose you may want to have something like `Arc<Mutex<Logger>>`, thus you will be able to copy this object among threads and benefit from automatic reference counting